include/net/net_pkt.h

```diff
@@ -72,6 +72,8 @@ struct net_pkt {
 	struct net_linkaddr lladdr_src;
 	struct net_linkaddr lladdr_dst;
 
+	u16_t data_len;         /* amount of payload data that can be added */
```

Comment:
I think this could be avoided, see below

Reply:
+1 . From my comment https://github.com/zephyrproject-rtos/zephyr/pull/119#issuecomment-339361679 :

> As far as I can tell, for TX packet, net_pkt->appdatalen isn't used until packet construction is finished and ready to be sent. So, it can be used to cache "remaining data size in the packet" (initialized in net_pkt_get_tx()).